internal/services/rewards.go

```diff
@@ -228,27 +185,34 @@ func (t *BaselineClient) assignPoints() error {
 			RewardsReceiverEthereumAddress: null.StringFrom(vOwner.Hex()),
 		}
 
-		validIntegrations := utils.NewSet(deviceActivity.Integrations...) // Guaranteed to be non-empty at this point.
+		userIntegrations := utils.NewSet(deviceActivity.Integrations...) // Guaranteed to be non-empty at this point.
+		userIntegrationPts := 0
 
-		if validIntegrations.Contains(integCalc.AutoPiID) {
-			if ud.AftermarketDeviceTokenId == nil {
-				validIntegrations.Remove(integCalc.AutoPiID)
-			} else {
-				thisWeek.AftermarketTokenID = types.NewNullDecimal(new(decimal.Big).SetUint64(*ud.AftermarketDeviceTokenId))
-
-				if len(ud.AftermarketDeviceBeneficiaryAddress) == 20 {
-					adBene := common.BytesToAddress(ud.AftermarketDeviceBeneficiaryAddress)
-					if vOwner != adBene {
-						logger.Info().Msgf("Sending tokens to beneficiary %s for aftermarket device %d.", adBene.Hex(), *ud.AftermarketDeviceTokenId)
-						thisWeek.RewardsReceiverEthereumAddress = null.StringFrom(adBene.Hex())
-					}
+		for _, integ := range integrations.Integrations {
```

Comment:
I think we talked about inverting this logic, so that we're not iterating over active integrations anymore. I think the result is going to be cleaner.